YellowNotes.Core/Repositories/NoteRepository.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Models;
+
+namespace YellowNotes.Core.Repositories
+{
+    public class NoteRepository : INoteRepository
+    {
+        private readonly DatabaseContext context;
+
+        public NoteRepository(DatabaseContext context) => this.context = context;
+
+        public async Task<bool> CreateNote(Note note, CancellationToken cancellationToken)
+        {
+            note.ModificationDate = DateTime.Now;
+            note.IsRemoved = false;
+
+            context.Notes.Add(note);
+            return await context.SaveChangesAsync(cancellationToken) > 0;
+        }
+
+        public async Task<IEnumerable<Note>> GetNotes(int count, string email,
+            CancellationToken cancellationToken)
+        {
+            return await context.Notes.Where(x => x.UserEmail == email && x.IsRemoved == false)
+                .OrderByDescending(x => x.ModificationDate)
+                .Take(count)
+                .AsNoTracking()
+                .ToListAsync(cancellationToken);
+        }
```

Comment:
Does this function always returns first couple of notes?

Reply:
It returns `count` notes or whole collection when `Notes.Count < count`.

Reply:
Ok, but what happens when user sends `n` requests to get all notes. Will he get whole collection or maybe just first `count` of notes but duplicated `n` times?

Reply:
The notes will be duplicated. I don't think it's a good idea to preserve the loading state between requests. I understand where you're coming from though. I'll think of a better solution.

Reply:
Some `startIndex` parameter would be cool.